src/gui/rss/automatedrssdownloader.cpp

```diff
@@ -421,17 +484,116 @@ void AutomatedRssDownloader::on_exportBtn_clicked()
+                }
+                for (int i = 0; i < filters->list_size(); ++i) {
+                    const lazy_entry *filter = filters->list_at(i);
+                    Rss::DownloadRulePtr rule(new Rss::DownloadRule());
+
+                    rule->setName(fromStdString(filter->dict_find_string_value("name")));
+                    qDebug() << "Importing \u03BCTorrent Rule: " << rule->name();
+
+                    rule->setEpisodeFilter(fromStdString(filter->dict_find_string_value("episode_filter")));
+                    rule->setSavePath(fromStdString(filter->dict_find_string_value("directory")));
+                    rule->setLastMatch(QDateTime::fromTime_t(filter->dict_find_int_value("last_match")));
+                    rule->setIgnoreDays(utorrent::POSTPONE_OPTS[filter->dict_find_int_value("postpone")]);
+
+                    int enabledState = filter->dict_find_int_value("enabled");
+                    rule->setEnabled(enabledState & utorrent::EnabledFlags::ENABLED);
+                    if (enabledState & utorrent::EnabledFlags::PAUSED)
+                        rule->setAddPaused(Rss::DownloadRule::AddPausedState::ALWAYS_PAUSED);
+
+                    int feedId = filter->dict_find_int_value("feed");
+                    if (feedId == -1) {
+                        QStringList urls;
+                        std::transform(importedFeedUrls.cbegin(), importedFeedUrls.cend(), std::back_inserter(urls), [](decltype(importedFeedUrls)::const_reference x){
+                            return x.second;
+                        });
+                        rule->setRssFeeds(urls);
+                    }
+                    else {
+                        rule->setRssFeeds(QStringList(importedFeedUrls[feedId]));
+                    }
+
+                    utorrent::applyFilter(rule, filter);
+                    m_editableRuleList->saveRule(rule);
+                }
+                manager->saveStreamList();
+                loadFeedList();
+
+                success = true;
+            }
+        }
+    }
+
+    return success;
+}
+
 void AutomatedRssDownloader::on_importBtn_clicked()
 {
+    #define QBITTORRENT_RULES "qBittorent (*.rssrules *.filters)"
+    #define UTORRENT_RULES "uTorrent (*.dat)"
+
     // Ask for filter path
-    QString load_path = QFileDialog::getOpenFileName(this, tr("Please point to the RSS download rules file"), QDir::homePath(), tr("Rules list") + QString(" (*.rssrules *.filters)"));
+    QString selectedFilter;
+    QString load_path = QFileDialog::getOpenFileName(this, tr("Please point to the RSS download rules file"), QDir::homePath(), QBITTORRENT_RULES ";;" UTORRENT_RULES, &selectedFilter);
     if (load_path.isEmpty() || !QFile::exists(load_path)) return;
-    // Load it
-    if (!m_editableRuleList->unserialize(load_path)) {
+
+    bool success = false;
+    if (selectedFilter == QBITTORRENT_RULES)
```

Comment:
> Will be very readable!!!

Indeed.

>  (I hope you understand that I don't think so)

How should I know?

> You save the lines?

With `if`: dangling variable declaration, ` = false` is meaningless, then when one  starts to read the `if` construction it's impossible to predict immediately what is inside. Frome the other hand, my one-liner shows from the very beginning that its is an initialisation, and that value depends on a condition. What functions are called to get the values is not so important. And, if you like, it can be written, for instance, as follows:
```C++
 bool  success = selectedFilter == QBITTORRENT_RULES ?  
                 m_editableRuleList->unserialize(load_path) : 
                 importFromUTorrent(load_path);
```
So I propose not to break declaration and initialisation.